editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/db/AppRepository.java

```diff
@@ -107,4 +119,50 @@ protected Void doInBackground(final InfoPanel... params) {
             return null;
         }
     }
+
+    /* TERRAIN FUNCTIONS ****************************************************************************************/
+
+    public void insertTerrain(Terrain terrain) {
+        new insertTerrainAsyncTask(mTerrainDao).execute(terrain);
+    }
+
+    private static class insertTerrainAsyncTask extends AsyncTask<Terrain, Void, Void> {
+
+        private TerrainDao mAsyncTaskDao;
+
+        insertTerrainAsyncTask(TerrainDao dao) {
+            mAsyncTaskDao = dao;
+        }
+
+        @Override
+        protected Void doInBackground(final Terrain... params) {
+            mAsyncTaskDao.insert(params[0]);
+            return null;
+        }
+    }
+
+    //Deletes a singular terrain from the database given its unique id
+    public void deleteTerrain(Terrain terrain) {
+        new deleteTerrainAsyncTask(mTerrainDao).execute(terrain);
+
+        // Gets all the flags that are associated with the terrain that was just deleted
+        // and deletes all these flags from the flag table
+        List<Flag> flagsInTerrain = mFlagDao.getFlagsInTerrainNotLive(terrain.getTerrainId());
+        for(Flag flag : flagsInTerrain) {
+            deleteFlag(flag);
```

Comment:
By adding `onDelete = CASCADE` to the `Flag` class, we can remove this manual flag cleanup code. See the previous comment.